### PR TITLE
Upgrade iOS 16 sleep data

### PR DESCRIPTION
**Pull request by srongch**

### CODE REVIEW

File: packages/health/ios/Classes/SwiftHealthPlugin.swift [hunk 1]
```diff
@@ -175,14 +178,17 @@ public class SwiftHealthPlugin: NSObject, FlutterPlugin {
...
-            let success = hasPermission(type: sampleType, access: permissions[index])
-            if (success == nil || success == false) {
-                result(success)
+            if let sampleType = dataTypeLookUp(key: type) {
+                let success = hasPermission(type: sampleType, access: permissions[index])
+                if (success == nil || success == false) {
+                    result(success)
+                    return
+                }
+            }else{
```
Comment (fix92) [inline packages/health/ios/Classes/SwiftHealthPlugin.swift]:
```suggestion
            } else {
```

File: packages/health/ios/Classes/SwiftHealthPlugin.swift [hunk 2]
```diff
@@ -724,13 +747,20 @@ public class SwiftHealthPlugin: NSObject, FlutterPlugin {
...
             dataTypesDict[WORKOUT] = HKSampleType.workoutType()
-            
             healthDataTypes = Array(dataTypesDict.values)
         }
+        
+        if #available(iOS 16.0, *) {
+            dataTypesDict[SLEEP_ASLEEP_CORE] = HKSampleType.categoryType(forIdentifier: .sleepAnalysis)!
+            dataTypesDict[SLEEP_ASLEEP_DEEP] = HKSampleType.categoryType(forIdentifier: .sleepAnalysis)!
+            dataTypesDict[SLEEP_ASLEEP_REM] = HKSampleType.categoryType(forIdentifier: .sleepAnalysis)!
+            //            dataTypesDict[SLEEP_ASLEEP_UNSPECIFIED] = HKSampleType.categoryType(forIdentifier: .sleepAnalysis)!
```
Comment (fix92) [inline packages/health/ios/Classes/SwiftHealthPlugin.swift]: I guess we should remove this comment then

**Review by fix92 — CHANGES_REQUESTED** [review]
two other things:
1. in the constructor at lib/src/health_data_point.dart:17 there are some special type conversions from seconds to minutes. Do we need this as well for the new types?
2. we should add some documentation in the README.md, so removing SLEEP_ASLEEP from ios and adding three entries for the new ones that are only available on ios and also add the comment that they require the watch

**Comment by srongch** [comment]
> two other things:
> 
> 1. in the constructor at lib/src/health_data_point.dart:17 there are some special type conversions from seconds to minutes. Do we need this as well for the new types?
> 2. we should add some documentation in the README.md, so removing SLEEP_ASLEEP from ios and adding three entries for the new ones that are only available on ios and also add the comment that they require the watch

1. Yes, we need that, I think. I updated it. 
2. Updated the README

**Review by fix92 — APPROVED** [review]

**Pull request merged**